### PR TITLE
blob: migrate gcsblob tests to use httpreplay instead of go-vcr

### PR DESCRIPTION
**Pull request by eliben**

### CODE REVIEW

File: internal/testing/setup/setup.go [hunk 1]
```diff
@@ -151,44 +151,51 @@ func NewAWSSession2(ctx context.Context, t *testing.T, region string) (sess *ses
...
 		if err != nil {
 			t.Fatal(err)
 		}
+		cleanup := func() {
+			if err := rec.Close(); err != nil {
+				t.Fatal(err)
+			}
+		}
+		return &gcp.HTTPClient{Client: *client}, client.Transport, cleanup
 	} else {
```
Comment (vangent) [inline internal/testing/setup/setup.go]: Nit: You can drop the `else` here because of the return above.

**Comment by vangent** [comment]
> setup.go:188: open testdata/TestPreconditions.replay: no such file or directory

I think that's a merge issue, it's a new test I just added.

File: internal/testing/setup/setup.go (hunk 1 above)
Comment (eliben) [reply internal/testing/setup/setup.go]: Done

**Comment by eliben** [comment]
> > setup.go:188: open testdata/TestPreconditions.replay: no such file or directory
> 
> I think that's a merge issue, it's a new test I just added.

Ugh... and this new test now fails when recording :-) Will investigate why

**Comment by eliben** [comment]
> > > setup.go:188: open testdata/TestPreconditions.replay: no such file or directory
> > 
> > 
> > I think that's a merge issue, it's a new test I just added.
> 
> Ugh... and this new test now fails when recording :-) Will investigate why

Aaaand it fails when recording in a clean client too

**Comment by eliben** [comment]
> > > > setup.go:188: open testdata/TestPreconditions.replay: no such file or directory
> > > 
> > > 
> > > I think that's a merge issue, it's a new test I just added.
> > 
> > 
> > Ugh... and this new test now fails when recording :-) Will investigate why
> 
> Aaaand it fails when recording in a clean client too

Merged @vangent 's fix in #1933. Travis rerunning tests now

**Pull request merged**